workflow/scripts/redact-page.py

```diff
@@ -25,108 +15,15 @@ def parse_page(
-        inplace=True,
-    )
-    detected_text_df.text = detected_text_df.text.str.lower()
-
-    return detected_text_df
-
-
-def select_personal_data(
-    detected_text_df: pd.DataFrame, personal_data: dict, min_distance: int
-) -> pd.DataFrame:
-    """Identifies personal data from the detected text.
-
-    Args:
-        detected_text_df (pd.DataFrame): detected text on the image.
-        personal_data (dict): personal data to be masked out
-        min_distance (int): maximum Levenshtein distance of the found text on the image to the personal data.
-
-    Returns:
-        pd.DataFrame: person data with location on image, filtered by min_distance.
-    """
-
-    final_df = pd.DataFrame()
-
-    max_spaces = max([value.count(" ") for value in personal_data.values()])
-    for no_spaces in range(max_spaces + 1):
-        tmp_df = detected_text_df.copy()
-        tmp_df.rename(columns={"text": "text_0", "width": "width_0"}, inplace=True)
-
-        # subset of personal data dict, according to # spaces
-        tmp_dict = {
-            key: value.replace(" ", "")
-            for key, value in personal_data.items()
-            if value.count(" ") == no_spaces
-        }
-
-        # shif text to get longer phrases
-        if no_spaces > 0:
-            for shift in range(1, no_spaces + 1):
-                # shift text column and aggreagte
-                highest_text_column_name = "text_" + str(shift)
-                tmp_df[highest_text_column_name] = tmp_df[
-                    "text_" + str(shift - 1)
-                ] + tmp_df.text_0.shift(-shift).fillna("")
-
-                # shift width column and aggreagte
-                highest_width_column_name = "width_" + str(shift)
-                tmp_df[highest_width_column_name] = (
-                    tmp_df["width_" + str(shift - 1)]
-                    + tmp_df.width_0.shift(-shift).fillna(0)
-                    + tmp_df.width_0 / tmp_df.text_0.str.len()
-                )
-
-            tmp_df["width_0"] = tmp_df[highest_width_column_name].astype(int)
-        else:
-            highest_text_column_name = "text_0"
+    if not ".jpg" in out_path[-3:]:
+        "".join([out_path, ".jpg"])
 
-        # calc edit distances for each key in subsampled dict
-        for key, value in tmp_dict.items():
-            tmp_df[key] = tmp_df[highest_text_column_name].apply(
-                lambda text: Levenshtein.distance(value, text)
-            )
-
-        tmp_df.rename(columns={"text_0": "text", "width_0": "width"}, inplace=True)
-        # select entries, where distance is below or equal a threshhold
-        query_list = []
-        for col in tmp_dict.keys():
-            query_list.append(f"{col}<={min_distance}")
-        tmp_df = tmp_df.query(" | ".join(query_list))
-
-        final_df = final_df.append(tmp_df, ignore_index=True)
-
-    return final_df[["left", "top", "width", "height", "conf", "text"]]
```

Comment:
Verstehe ich es richtig, dass du erst einmal eine vereinfachte minimal Version testen willst, in der nur perfect matches geschwärzt werden, um zu sehen ob es durchläuft?